js/screens/Login/LoginScreen.js

```diff
@@ -34,29 +34,24 @@ class LoginContainer extends Component {
   handlePassword(text) {
     this.props.dispatch(fetchPassword(text));
   }
-  _signInAsync = async () => {
-    await AsyncStorage.setItem('userToken', 'abc');
+  _signInAsync = async email => {
+    await AsyncStorage.setItem('userToken', email);
     this.props.navigation.navigate('LocationSearch');
   };
   handleSubmit() {
     const { email, password } = this.props;
 
-    const errorMessage = {
-      code: 'Invalid login',
-      message: 'Please Enter a Valid Email Address'
-    };
+    const signInSuccess = isValidEmailAndPassword(email, password);
 
-    if (isValidEmailAndPassword(email, password)) {
-      firebaseAuth
-        .signInWithEmailAndPassword(email, password)
-        .then(this._signInAsync)
+    signInSuccess
+      ? firebaseAuth
+          .signInWithEmailAndPassword(email, password)
+          .then(this._signInAsync(email))
```

Comment:
When you get a promise back from the firebase you get a user info in your response. so it should be .then((user) => this._signInAsync(user)) not email and using this, store user info in your asyncStorage, not an email.

Please make this change